feat(resources): add deprecation for resources whose name breaks RFC-1035 (#13003)
kumahq/kuma: UPGRADE.md

Before the upgrade ensure that your resources don't use unsupported characters.

> Further, starting from version `2.10.x` we are deprecating the usage of non [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035.html) characters for names of `Mesh`, `Zone`, `MeshService`, `MeshExternalService`, `MeshMultizoneService` resources. These names will not be accepted in a future release. 

Reviewer: In the rule above there are no character limit?
```suggestion
Starting from version `2.10.x` we are deprecating the usage of non [RFC 1035](https://www.rfc-editor.org/rfc/rfc1035.html) characters for names of `Mesh`, `Zone`, `MeshService`, `MeshExternalService`, `MeshMultizoneService` resources. These names will be rejected in the future.
```
